werkzeug/contrib/cache.py

```diff
@@ -658,7 +674,7 @@ def _prune(self):
                     remove = False
                     with open(fname, 'rb') as f:
                         expires = pickle.load(f)
-                    remove = expires <= now or idx % 3 == 0
+                    remove = (expires != 0 and expires <= now) or idx % 3 == 0
```

Comment:
I did take this out, but ended up putting it back in since expires is zero in the case where you do not want it expire at all. Else when expires is zero the value will always be pruned.